Assets/Scripts/ActionMaster.cs

```diff
@@ -1,48 +0,0 @@
-﻿using System;
-using UnityEngine;
-using System.Collections;
-using System.Collections.Generic;
-using System.Linq;
-using System.Resources;
-
-public class ActionMaster
-{
-	
-	public enum Action
-	{
-		MidFrameReset,
-		FrameReset,
-		EndTurn,
-		EndGame
-	};
-
-	public Action NextAction(Frame frame)
-	{
-		var lastRoll = frame.GetRolls().Last();
-
-		if (lastRoll < 0 || lastRoll > 10)
-		{
-			throw new UnityException("Invalid pin count");
```

Comment:
Good that you're looking for invalid states here, and throwing exceptions.

The UnityException is however, not the correct exception type. There are lots, and each has a specific purpose. This one is for internal Unity errors.

For this use case you want to use the [ArgumentException](https://msdn.microsoft.com/en-us/library/system.argumentexception(v=vs.110).aspx) which is a regular C# exception, and used for when you're calling a method with an argument which isn't correctly configured.

As this is one of the outputs of the ActionMaster I'd also recommend you test for this. It's important to think about the failure, as well as success states when writing tests:

```
[Test]
public void testNoFramesThrowsArgumentException()
{
    // Deliberately add no rolls.
    currentFrame = new Frame(new int[] {});

    Assert.Throws<ArgumentException>(() => {
        actionMaster.NextAction(currentFrame);
    });
}

[Test]
public void testMoreThan10RollsThrowsArgumentException()
{
   ...
}
````